prusti-encoder/src/encoders/type/domain.rs

```diff
@@ -194,22 +255,42 @@ impl TaskEncoder for DomainEnc {
                 TyKind::Str => super::kinds::str::domain(*task_key, deps, builder)?,
                 _kind => super::kinds::opaque::domain(*task_key, deps, builder)?,
             };
-            Ok((PureTypeCommon::build(builder), specifics))
+            Ok((PureTypeCommon::build(builder), DomainEncOutput { inner: output_ref, specifics }))
         })
     }
+
+    fn emit_outputs<'vir>(program: &mut task_encoder::Program<'vir>) {
+        for output in DomainEnc::all_outputs_local() {
+            program.add_function(output.unreachable_to_snap);
+            match output.kind {
+                DomainEncLocalKind::Domain {
+                    domain
+                } => program.add_domain(domain),
+                DomainEncLocalKind::Adt { adt, discr_fn } => {
+                    program.add_adt(adt);
+                    if let Some(discr_fn) = discr_fn {
+                        program.add_function(discr_fn);
+                    }
+                }
+                DomainEncLocalKind::None => {}
+            }
+        }
+    }
 }
 
 pub(crate) struct PureTypeCommon<'vir> {
     pub(crate) vcx: &'vir vir::VirCtxt<'vir>,
     name: Option<&'vir str>,
     generics: Option<Vec<vir::LocalDeclTyVal<'vir>>>,
-    domain_ident: Option<vir::DomainIdnCSnap<'vir>>,
-    self_type: Option<vir::TypeCSnap<'vir>>,
+    domain_ident: Option<vir::DomainIdnSnap<'vir>>,
+    self_type: Option<vir::TypeSnap<'vir>>,
+    unreachable_to_snap: Option<FunctionIdn<'vir, (), vir::Snap>>,
 }
 
 pub(crate) struct DomainBuilder<'vir> {
     axioms: Vec<vir::DomainAxiom<'vir>>,
     functions: Vec<vir::DomainFunction<'vir>>,
+    // e_functions: Vec<vir::Function<'vir>>,
```

Comment:
`e_functions` as in... extra? functions? Do you foresee the need for this kind of thing? If not, then this + the method to add functions (also commented out) can all be removed.